app/javascript/react/components/molecules/FixedStreamStationHeader/atoms/StationActionButtons/StationActionButtons.tsx

```diff
@@ -2,21 +2,38 @@ import React from "react";
 import { useTranslation } from "react-i18next";
 
 import copyLink from "../../../../../assets/icons/copyLink.svg";
-import shareLink from "../../../../../assets/icons/shareLink.svg";
 import downloadImage from "../../../../../assets/icons/download.svg";
-import { Button } from "../../../../Button/Button.style";
+import shareLink from "../../../../../assets/icons/shareLink.svg";
+import { copyCurrentURL } from "../../../../../utils/copyCurrentUrl";
+
+import useShortenedLink from "../../../../../utils/urlShortenedLink";
 import { ActionButton } from "../../../../ActionButton/ActionButton.style";
-import * as S from "./StationActionButtons.style";
+import { Button } from "../../../../Button/Button.style";
 import { ExportDataModal } from "../../../../Modals";
-import { copyCurrentURL } from "../../../../../utils/copyCurrentUrl";
 import { SmallPopup } from "../../../../Modals/SessionDetailsModal/SessionDetailsModal.style";
+import * as S from "./StationActionButtons.style";
 
 interface Props {
   sessionId: string;
 }
+const BITLY_ACCESS_TOKEN = process.env.BITLY_ACCESS_TOKEN || "";
 
 const StationActionButtons = ({ sessionId }: Props) => {
+  const currentUrl = window.location.href;
   const { t } = useTranslation();
+  const { shortenedLink, error } = useShortenedLink(
+    currentUrl,
+    BITLY_ACCESS_TOKEN
+  );
+
+  const handleCopyLink = () => {
+    copyCurrentURL(shortenedLink);
+    alert("Link copied to clipboard");
+  };
+
+  if (error) {
+    console.error("Error shortening link: ", error.message);
+  }
```

Comment:
It could be a good idea to add a null check for `shortenedLink ` while handling the copy link and if it fails I'd show a user aan alert that it can't be copied

We could take advantage of translations in case of alert messages.